File: p4/DevoradorTiempos.java
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;
import java.util.*;
import org.json.simple.JSONObject;
import org.json.simple.parser.JSONParser;
import org.json.simple.parser.ParseException;

public class DevoradorTiempos {
	public static void main(String[] args) {
		JSONParser parser = new JSONParser();
		for(int n = 8;n<=65536;n*=2){
			String archivo = "sols/g" + n + ".json";
			try (FileReader reader = new FileReader(archivo)) {
				JSONObject jsonObject = (JSONObject) parser.parse(reader);
				@SuppressWarnings("unchecked")
				Map<String, List<String>> grafo = (Map<String, List<String>>) jsonObject.get("grafo");

				long inicio = System.currentTimeMillis();
                Map<String, String> solucion = ColoreoGrafo.realizarVoraz(grafo);
                long fin = System.currentTimeMillis();

                double tiempoMs = (fin - inicio);
                System.out.println("Grafo: " + archivo + " -> Tiempo: " + tiempoMs + " ms");
                //System.out.println("Colores usados: " + new HashSet<>(solucion.values()).size());
				
			} catch (IOException | ParseException e) {
				e.printStackTrace();
			}
		}
	}
}

File: p4/devorador_tiempos.py
import time
import json
from typing import Any, Dict
from coloreado_grafo import realizar_voraz

def cargar_grafo_desde_json(archivo: str) -> Dict[str, Any]:
    with open(archivo, 'r') as f:
        grafo = json.load(f)
    return grafo

if __name__ == "__main__":

    n= 8
    while n <= 65537: 
        archivo_grafo = f"sols/g{n}.json"  
        try:
            mapa = cargar_grafo_desde_json(archivo_grafo)
            
            inicio = time.time()  
            solucion = realizar_voraz(mapa["grafo"])  
            fin = time.time()  
            
            tiempo = (fin - inicio) * 1000
            print(f"Tiempo para grafo de tamaño {n}: {tiempo:.2f} milisegundos")
        
        except FileNotFoundError:
            print(f"No se encontró el archivo para el grafo de tamaño {n}.")
            continue
        n*=2

